src/reconstruction/createTissueSpecificModel.m

```diff
@@ -271,53 +308,39 @@
         Rxns.UnknownIncluded = tissueModel.rxns(loc);
         
     case 'GIMME'
-        x = ismember(model.rxns,[ExpressedRxns;UnExpressedRxns]);
-        unk = find(x==0);
-        
-        expressionCol = zeros(length(model.rxns),1);
-        for i = 1:length(unk)
-            expressionCol(unk(i)) = -1;
-        end
-        
-        for i = 1:length(RHindex)
-            expressionCol(RHindex(i)) = 2;
-        end
         if ~exist('options','var') || isempty(options)
             loc = find(model.c);
             
             sol = optimizeCbModel(model);
             
             options = [loc 0.9];
         end
-        cutoff = 1;
-        [reactionActivity,reactionActivityIrrev,model2gimme,gimmeSolution] = solveGimme(model,options,expressionCol,cutoff);
+        [reactionActivity,reactionActivityIrrev,model2gimme,gimmeSolution, rxnFluxRev] = solveGimme(model,options,expressionCol,cutoff);
         
-        remove = model.rxns(find(reactionActivity == 0));
+        remove = model.rxns(reactionActivity == 0);
         tissueModel = removeRxns(model,remove);
-        
-        if funcModel ==1
-            c = tissueModel.c;
-            
-            remove = [];
-            tissueModel.c = zeros(length(tissueModel.c),1);
-            for i = 1:length(tissueModel.rxns)
-                tissueModel.c(i) = 1;
-                sol1 = optimizeCbModel(tissueModel,'max');
-                sol2 = optimizeCbModel(tissueModel,'min');
-                if sol1.f == 0 & sol2.f == 0
-                    remove = [remove tissueModel.rxns(i)];
-                end
-                tissueModel.c(i) = 0;
-            end
-            
-            tissueModel.c = c;
-            tissueModel = removeRxns(tissueModel,remove);
+        rxnFluxRev = rxnFluxRev(reactionActivity ~= 0);
+
+        if funcModel == 1
+            [minFlux, maxFlux] = fluxVariability(tissueModel, 90, 'max', tissueModel.rxns, 1);
+            remove = minFlux == 0 & maxFlux == 0;
+            rxnFluxRev = rxnFluxRev(~remove);
+            tissueModel = removeRxns(tissueModel, tissueModel.rxns(remove));
         end
         
+        SolutionFlux.Rev = rxnFluxRev;
+        SolutionFlux.Irrev = gimmeSolution.x;
+        
         Rxns.Expressed = ExpressedRxns;
         Rxns.UnExpressed = UnExpressedRxns;
         Rxns.unknown = unknown;
         
+        % These lines are faster and neater, but will give different order for the

```

Comment:
This section has alternative option. I think that before committing we can decide if yes or no, and finalize it or just erase them.